### ISSUE TITLE
Object.dirname is deprecated. ( is`file` variable a string? ) 

### ISSUE THREAD

**Issue by JeffBusterCase** [AtomLinter/linter-rust#101]
Argument to `path.dirname` must be a string
```
Object.dirname (/usr/share/atom/resources/app.asar/src/electron-shims.js:9:10)
<unknown> (/home/jeffbustercase/.atom/packages/linter-rust/lib/linter-rust.coffee:97:25)
```

I don't know where is `file` var from, but is giving erros on the current update of atom. After a little time of program or at the start, reports as deprecated package.

thanks for the attention

**Comment by Arcanemagus** [comment]
@gmist If I'm following the code correctly, this could only be triggered if `textEditor.getPath()` returned something other than a string, or in other words if a lint was triggered on a file that had never been saved.

**Comment by JeffBusterCase** [comment]
If I get what are you saying, no, the file was saved. Yet right about `textEditor.getPath()`, really points to that.

**Comment by Arcanemagus** [comment]
Unfortunately this project is still on CoffeeScript, so the line number in your trace is useless so I'm not sure which of the `dirname` calls triggered this.

**Comment by JeffBusterCase** [comment]
Sorry, but I'm starting to believe this is a bug of atom. Now that he called as deprecated the atom core itself , on `electron-shims.js` again

**Comment by cata0309** [comment]
Did you solved this problem ? The bug still exists in the latest version

**Comment by Arcanemagus** [comment]
@cata0309 Can you please file a new issue here with steps to reproduce? From what I can see there _should_ be checks that the paths are defined strings before any call to `dirname` so the issue as above is fixed.
